From 434bc8452010efcefbcd039f605ba0c53c42757f Mon Sep 17 00:00:00 2001
From: Eric Sauvageau <merlin@asuswrt-merlin.net>
Date: Fri, 3 Jul 2026 12:15:52 -0400
Subject: [PATCH] rc: limit ipv6 LAN prefix to a /64 on non-multilan platforms

Similar to the previous behaviour, but still accept weird prefixes
that are shorter than a /64.

(cherry picked from commit a89e9ecaf819c7cf82ebc340b6c103f029a4fd90)
---
 release/src/router/rc/services.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/release/src/router/rc/services.c b/release/src/router/rc/services.c
index e6b5ef8d26..55c5521c3a 100644
--- a/release/src/router/rc/services.c
+++ b/release/src/router/rc/services.c
@@ -2869,9 +2869,7 @@ void add_ip6_lanaddr(void)
 	p = ipv6_router_address(NULL);
 	if (*p) {
 		prefix_len = nvram_get_int(ipv6_nvname("ipv6_prefix_length"));
-		if (prefix_len <= 56)
-			prefix_len = 64;
-		snprintf(ip, sizeof(ip), "%s/%d", p, prefix_len);
+		snprintf(ip, sizeof(ip), "%s/%d", p, (prefix_len > 64 ? prefix_len : 64));
 		eval("ip", "-6", "addr", "add", ip, "dev", nvram_safe_get("lan_ifname"));
 		if (!nvram_match(ipv6_nvname("ipv6_rtr_addr"), (char*)p))
 			nvram_set(ipv6_nvname("ipv6_rtr_addr"), (char*)p);
-- 
2.25.1

